engines/dfc_provider/app/controllers/dfc_provider/enterprise_groups/affiliated_by_controller.rb

```diff
@@ -22,6 +22,14 @@ def create
         head :created
       end
 
+      def destroy
+        group = EnterpriseGroup.find(params[:enterprise_group_id])
+
+        authorize! :update, group
```

Comment:
I was about to say this should be `:destroy`, but we are destroying the relationship, not the group itself. So `:update` is the right choice :)